integration-tests/chain-signatures/tests/cases/mod.rs

```diff
@@ -399,162 +363,123 @@ fn dummy_triple(id: u64) -> Triple {
-            // up the new contract by first upgrading the contract, then trying to generate a new signature.
-            let id = ctx.propose_update_contract_default().await;
-            ctx.vote_update(id).await;
-            tokio::time::sleep(std::time::Duration::from_secs(3)).await;
-            wait_for::has_at_least_mine_presignatures(&ctx, 1).await?;
-            actions::single_payload_signature_production(&ctx, &state).await?;
-
-            // Now do a config update and see if that also updates the same:
-            let id = ctx
-                .propose_update(ProposeUpdateArgs {
-                    code: None,
-                    config: Some(Config::default()),
-                })
-                .await;
-            ctx.vote_update(id).await;
-            tokio::time::sleep(std::time::Duration::from_secs(3)).await;
-            wait_for::has_at_least_mine_presignatures(&ctx, 1).await?;
-            actions::single_payload_signature_production(&ctx, &state).await?;
-
-            Ok(())
+    let nodes = cluster::spawn().wait_for_running().await?;
+    nodes.wait().signable().await?;
+    nodes.sign().await.unwrap();
+
+    // Perform update to the contract and see that the nodes are still properly running and picking
+    // up the new contract by first upgrading the contract, then trying to generate a new signature.
+    let id = nodes.propose_update_contract_default().await;
+    nodes.vote_update(id).await;
+    nodes.wait().signable().await?;
+    nodes.sign().await.unwrap();
+
+    // Now do a config update and see if that also updates the same:
+    let id = nodes
+        .propose_update(ProposeUpdateArgs {
+            code: None,
+            config: Some(Config::default()),
         })
-    })
-    .await
+        .await;
+    nodes.vote_update(id).await;
+    nodes.wait().signable().await?;
+    nodes.sign().await.unwrap();
+
+    Ok(())
 }
 
 #[test(tokio::test)]
 async fn test_batch_random_signature() -> anyhow::Result<()> {
-    with_multichain_nodes(MultichainConfig::default(), |ctx| {
-        Box::pin(async move {
-            let state_0 = wait_for::running_mpc(&ctx, Some(0)).await?;
-            assert_eq!(state_0.participants.len(), 3);
-            actions::batch_random_signature_production(&ctx, &state_0).await?;
-            Ok(())
-        })
-    })
-    .await
+    let nodes = cluster::spawn().wait_for_running().await?;
+    actions::batch_random_signature_production(&nodes).await?;
+    Ok(())
 }
 
 #[test(tokio::test)]
 async fn test_batch_duplicate_signature() -> anyhow::Result<()> {
-    with_multichain_nodes(MultichainConfig::default(), |ctx| {
-        Box::pin(async move {
-            let state_0 = wait_for::running_mpc(&ctx, Some(0)).await?;
-            assert_eq!(state_0.participants.len(), 3);
-            actions::batch_duplicate_signature_production(&ctx, &state_0).await?;
-            Ok(())
-        })
-    })
-    .await
+    let nodes = cluster::spawn().wait_for_running().await?;
```

Comment:
Are we getting nodes or cluster here?

Reply:
`nodes` is type of ` Cluster` here since they're both equivalent